libparsec/crates/client/tests/unit/workspace_inbound_sync_monitor.rs

```diff
@@ -0,0 +1,436 @@
+                .field("outcome", outcome)
+                .finish(),
+            InboundSyncMonitorEvent::RetryLaterBusyEntry {
+                expected_entry_id, ..
+            } => f
+                .debug_struct("RetryLaterBusyEntry")
+                .field("expected_entry_id", expected_entry_id)
+                .finish(),
+            Self::EventBusWaitServerOnline { .. } => {
+                f.debug_struct("EventBusWaitServerOnline").finish()
+            }
+            Self::EventBusSend { .. } => f.debug_struct("EventBusSend").finish(),
+            Self::WorkspaceOpsRefreshRealmCheckpoint { outcome, .. } => f
+                .debug_struct("RefreshRealmCheckpoint")
+                .field("outcome", outcome)
+                .finish(),
+            Self::WorkspaceOpsGetNeedInboundSync {
+                expected_limit,
+                outcome,
+                ..
+            } => f
+                .debug_struct("GetNeedInboundSync")
+                .field("expected_limit", expected_limit)
+                .field("outcome", outcome)
+                .finish(),
+            Self::WorkspaceOpsInboundSync {
+                expected_entry_id,
+                outcome,
+                ..
+            } => f
+                .debug_struct("InboundSync")
+                .field("expected_entry_id", expected_entry_id)
+                .field("outcome", outcome)
+                .finish(),
+        }
+    }
+}
+
+struct MockedInboundSyncManagerIOInternal {
+    expected_events: VecDeque<InboundSyncMonitorEvent>,
+    abort_task: Option<Box<dyn FnOnce() + Send + 'static>>,
+}
+
+struct MockedInboundSyncManagerIO {
+    internal: Mutex<MockedInboundSyncManagerIOInternal>,
+}
+
+impl MockedInboundSyncManagerIO {
+    async fn pop_next_expected_event(&self) -> InboundSyncMonitorEvent {
+        let next_expected_event = {
+            let mut internal = self.internal.lock().expect("Mutex is poisoned");
+            internal.expected_events.pop_front()
+        };
+        match next_expected_event {
+            Some(next_expected_event) => next_expected_event,
+            None => {
+                {
+                    let mut internal = self.internal.lock().expect("Mutex is poisoned");
+                    let abort_task = internal
+                        .abort_task
+                        .take()
+                        .expect("`abort_task()` already called !");
+                    abort_task();
+                }
+                // Abort is a fire-and-forget call, we then sleep forever and wait for our
+                // task to actually get aborted.
+                libparsec_platform_async::sleep(std::time::Duration::from_secs(u64::MAX)).await;
+                unreachable!()
+            }
+        }
+    }
+}
+
+impl InboundSyncManagerIO for MockedInboundSyncManagerIO {
+    async fn wait_for_next_incoming_event(&self) -> Result<super::IncomingEvent, ()> {
+        println!(">>> wait_for_next_incoming_event()");
```

Comment:
I'm against using log in the tests: 
- most test don't need to display anything.
- when a test displays something, it's because it is really important when debugging those tests.

Given the output is not shown when the test succeed, it seems more convenient to have those print displayed by default given that's what you want when the test fail.

On the other hand using log means we must pass `RUST_LOG=DEBUG`, but then we also have enabled all the others logs we don't care about, so now we have to find the path of the test module we want to debug to further customize `RUST_LOG`'s config...